Allow `containerd` config imports (#1630)
awslabs/amazon-eks-ami: doc/USER_GUIDE.md

---

## Customizing Containerd Config

> In some cases, customers may want to customize the [containerd configuration](https://github.com/containerd/containerd/blob/main/docs/man/containerd-config.toml.5.md)
> on their nodes. We utilize the drop-in style `import` files [supported by containerd](https://github.com/containerd/containerd/blob/main/docs/man/containerd-config.toml.5.md#format)
> to achieve this. 
> 
> **Import a containerd config**
> 
> If you have a containerd config of your own, you can place it inside the directory `/etc/containerd/config.d/` via your user data.
> Make sure that you explicitly set the `.toml` extension on the file.

Reviewer: ```suggestion
The EKS defaults for `containerd` will be written to `/etc/containerd/config.toml`. Additional configuration files placed in the `/etc/containerd/config.d/` directory will be imported by the default configuration. Additional configuration files will override the default as described in the [`containerd` documentation](https://github.com/containerd/containerd/blob/release/1.7/docs/man/containerd-config.toml.5.md).
```

> > [!NOTE]
> > When adding a new `containerd` config file to the `import` path after `containerd` has already started (i.e. `bootstrap.sh` already ran),
> > you'll need to restart `containerd` to pick up the latest configuration.

Reviewer: ```suggestion
> [!NOTE]
> If you create an additional configuration file after `containerd` has already started (i.e. `bootstrap.sh` has already executed),
> you'll need to restart `containerd` to pick up the latest configuration.
```

> The `bootstrap.sh` script will write the defaults for the containerd config to `/etc/containerd/config.toml`, containing an `import` 
> list containing the path glob: `/etc/containerd/config.d/*.toml`. `bootstrap.sh` will also create this directory,
> so the user is not required create it when executing commands after the bootstrap is run.

Reviewer: Don't think we need to explain all this here, but lets make sure that directory is created when the AMI is built, not by the bootstrap script.
```suggestion
```

Author: i don't want to omit any info about the pattern/file-ext, Ill just change the pattern to `/etc/containerd/config.d/*` and then ill make this change

Reviewer: Sure, I don't feel too strongly about that detail, but I don't think anyone would expect files without a `.toml` extension to be included, all containerd config files are TOML files.

> > [!IMPORTANT]
> > containerd `imports` is the recommended approach to configure contianerd on EKS Optimized AMIS.  
> > Please attempt to use this method before ever making direct edits to `/etc/containerd/config.toml`.

Reviewer: ```suggestion
> [!CAUTION]
> Making direct edits to the EKS default `containerd` configuration file is not recommended.
```

   
**View active containerd config**

To see the final configuration that is produced and consumed by containerd, you
can use the containerd cli:
```
$ containerd config dump

> 
> version = 2
> root = "/var/lib/containerd"
> state = "/run/containerd"
> oom_score = 0
> imports = ["/etc/containerd/config.d/*.toml"]
> 
> [grpc]
>   address = "/run/containerd/containerd.sock"
>   uid = 0
>   gid = 0
> 
> [debug]
>   address = "/run/containerd/debug.sock"
>   uid = 0
>   gid = 0
>   level = "info"

Reviewer: ```suggestion
```
